ccgains/historic_data.py

```diff
@@ -415,3 +415,232 @@ def prepare_request(self, dtime):
+        saved in *cache_folder*.
+
+        The *unit* must be a string given in the form
+        'currency_one/currency_two', e.g. 'BTC/USD'.
+
+        The data will be resampled by calculating the weighted price
+        for interval steps specified by *interval*. See:
+        http://pandas.pydata.org/pandas-docs/stable/timeseries.html#offset-aliases
+        for possible values.
+        """
+
+        super(HistoricDataAPICoinbase, self).__init__(unit)
+        self.interval = interval
+        self.url = 'https://api.pro.coinbase.com/products'
+        self.command = "/candles"
+
+        # Coinbase Pro rate limit is 3 per second
+        self.query_wait_time = 0.334
+
+        self.currency_pair = '{0.cfrom:s}-{0.cto:s}'.format(self)
+
+        self.connection_error = requests.ConnectionError(
+            'Price data for %s could not be loaded from %s '
+            '- are you online?' % (self.currency_pair, self.url))
+
+        file_name = path.join(
+            cache_folder,
+            'Coinbase_{0.cfrom:s}-{0.cto:s}_{0.interval:s}.h5'.format(self))
+        flipped_file_name = path.join(
+            cache_folder,
+            'Coinbase_{0.cto:s}-{0.cfrom:s}_{0.interval:s}.h5'.format(self))
+
+        if path.exists(file_name):
+            self.file_name = file_name
+        elif path.exists(flipped_file_name):
+            # Flip currency pair
+            self.cfrom, self.cto = self.cto, self.cfrom
+            self.unit = self.cto + '/' + self.cfrom
+            self.file_name = flipped_file_name
+            self.currency_pair = '{0.cfrom:s}-{0.cto:s}'.format(self)
+        else:
+            try:
+                req = requests.get(self.url)
+            except requests.ConnectionError:
+                raise self.connection_error
+
+            known_symbols = [symbol['id'] for symbol in req.json()]
+
+            if self.currency_pair in known_symbols:
+                self.file_name = file_name
+            else:
+                # Try flipped currency pair
+                flipped_currency_pair = '{0.cto:s}-{0.cfrom:s}'.format(self)
+                if flipped_currency_pair not in known_symbols:
+                    raise ValueError(
+                        'Neither currency pair"{0:s}" nor pair "{1:s}" is '
+                        'available on "{2:s}".'.format(
+                            self.currency_pair, flipped_currency_pair, self.url))
+
+                # Flip currency pair
+                self.cfrom, self.cto = self.cto, self.cfrom
+                self.unit = self.cto + '/' + self.cfrom
+                self.currency_pair = flipped_currency_pair
+                self.file_name = flipped_file_name
+
+    def _wait_if_needed(self):
+        # Wait for minimum call time to pass:
+        now = pd.Timestamp.now()
+        delta = (now - self.last_query_time).total_seconds()
+
+        log.info(f'Coinbase last_query_time: {self.last_query_time}')
```

Comment:
This seems very verbose, logging every one of Coinbase's query's times. Did you maybe just put it here while debugging and forgot to remove it?

Reply:
Yup, just a missed debugging statement. Thanks. Will fix this and the others tonight and resubmit